src/Symfony/Bridge/Doctrine/Form/DataTransformer/CollectionToArrayTransformer.php

```diff
@@ -40,8 +41,10 @@ public function transform(mixed $collection): mixed
             return $collection;
         }
 
-        if (!$collection instanceof Collection) {
+        if (!interface_exists(ReadableCollection::class) && ! $collection instanceof Collection) {
```

Comment:
We need to make a decision here. Either we build compat logic for Collections 1.0 or we bump to Collections 1.8. Doing both at the same time is a bit pointless.

Reply:
Any meaningful test will need to reference the interface, so if we want a test, we need to bump the (dev) dependency, don't we?

I'm not sure of the best approach here. Trilemma seems to be:

* Bump the dependency, avoid the need for compat logic, and keep the test (weak as it is), or
* Include the compat logic, omit the test, don't bump, or
* Include compat logic, include the test, bump (current state of the PR).

Reply:
Option 4: Skip the test if the interface doesn't exist.

That being said, I'd go for your option 1. That compat code is an unnecessary liability.

Reply:
I'm okay with option 1 but, if I'm not wrong, we'd have to make the dependency mandatory.

Reply:
> Or, I suppose we could `conflict` < 1.8.0 out, as @xabbuh suggests below?

This is the way. (Insert Mandalorian GIF here.)

Reply:
Indeed. I think we should be good to go now?